src/PayID/pay-id-client.ts

```diff
@@ -0,0 +1,31 @@
+import PayIDError from './pay-id-error'
+
+/* eslint-disable @typescript-eslint/require-await */
+// TODO(keefertaylor): Enable lint rules when method is implemented.
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable max-classes-per-file */
+
+/**
+ * A client for PayID.
+ *
+ * Inputs to this class are always in X-Addresses.
+ * @see https://xrpaddress.info/
+ *
+ * @warning This class is experimental and should not be used in production applications.
+ * TODO(keefertaylor): Export this class in index.ts when it's ready for external consumption.
+ */
+export default class PayIDClient {
+  /**
+   * Retrieve the XRP Address authorized with a PayID.
+   *
+   * @note The returned value will always be in an X-Address format.
+   *
+   * @param payID The payID to resolve for an address.
+   * @returns An XRP address representing the given PayID if one exists, otherwise undefined.
```

Comment:
You may want to add a (optional?) `@param network` here. You can default to `MAINNET` in your code, or you can not pass a `Network` header at all and it will (currently) default to returning the associated Mainnet address.

TBD, @austin-king and I need to decide if `Network` is a mandatory or optional header when requesting a PayID, and whether it's a part of the service or the core protocol. I personally think that an optional header is more user-friendly, but perhaps a mandatory header makes for a simpler API and ensures it doesn't become a hidden feature.

Reply:
Is the network in the URL + Content-Type tuple?

Reply:
Right now we've been playing with a separate (optional?) HTTP header.

So a request looks like:
```
GET https://xpring.io/hbergren
 Accept: "application/xrp+json"
 Network: "Testnet"
```

Currently if network isn't defined, it defaults to Mainnet.

Reply:
Deciding on final format with @sappenin later today.